components/chassis.py

```diff
@@ -35,40 +36,44 @@
 from utilities.position import TeamPoses
 
 
-class SwerveModule:
-    DRIVE_GEAR_RATIO = (14.0 / 50.0) * (25.0 / 19.0) * (15.0 / 45.0)
-    STEER_GEAR_RATIO = (14 / 50) * (10 / 60)
-    WHEEL_CIRCUMFERENCE = 4 * 2.54 / 100 * math.pi
+@dataclass
+class SwerveConfig:
+    drive_ratio: float
+    drive_gains: Slot0Configs
+    steer_ratio: float
+    steer_gains: Slot0Configs
+    reverse_drive: bool
+    wheel_circumference: float = 4 * 2.54 / 100 * math.pi
 
-    DRIVE_MOTOR_REV_TO_METRES = WHEEL_CIRCUMFERENCE * DRIVE_GEAR_RATIO
-    STEER_MOTOR_REV_TO_RAD = math.tau * STEER_GEAR_RATIO
 
+class SwerveModule:
     # limit the acceleration of the commanded speeds of the robot to what is actually
     # achiveable without the wheels slipping. This is done to improve odometry
     accel_limit = 15  # m/s^2
 
+    config: SwerveConfig
+
     def __init__(
         self,
-        x: float,
-        y: float,
+        config: SwerveConfig,
+        position: Translation2d,
         drive_id: int,
         steer_id: int,
         encoder_id: int,
-        *,
-        drive_reversed: bool = False,
     ):
+        self.config = config
```

Comment:
I don't see this being read anywhere outside `__init__`. Why does it need to be saved?

Also, this line is being inserted before the docstring, which makes it no longer a docstring, which is incorrect.

Reply:
that was a remnant of me trying to make a function to replace `DRIVE_MOTOR_REV_TO_METRES` but then realising i could do that in `ChassisComponent`

sorry abt the docstring thing lol